### PR TITLE
[[ Bug 21250 ]] Fix crash which may occur when selected objects are deleted

### PR DESCRIPTION
**Pull request by livecodeian**

### CODE REVIEW

File: engine/src/sellst.cpp [hunk 1]
```diff
@@ -133,7 +133,12 @@ void MCSellist::Clean()
             MCSelnode* t_next = t_cursor->next();
             t_cursor->remove(objects);
             delete t_cursor;
-            t_cursor = t_next;
+            if (objects == nil)
+            	t_cursor = nil;
+			else if (objects == t_next)
+				t_cursor = objects->next();
```
Comment (livecodeali) [inline engine/src/sellst.cpp]: Why do we have to loop round again in this situation?

Comment (livecodeian) [reply engine/src/sellst.cpp]: I may have to rethink this - we need to detect when the first object has been removed, so have to check if `objects` is now null (or now points to the next object) but make sure the loop still continues if there is now just a single object remaining (which hasn't been checked yet)
Basically, iterating over a double linked list while removing items from it is a pain :(

**Comment by livecodeali** [comment]
Heh, OK I understand this much better :-D
@livecode-vulcan review ok f665463

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok f665463eb7e702846e1d0694f27cdf290194b0a0

**Comment by livecode-vulcan** [comment]
😎 test success f665463eb7e702846e1d0694f27cdf290194b0a0

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1482)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1498)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1472)
* try-community-universal-ios-iphoneos11.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.3/builds/75)
* try-community-universal-ios-iphonesimulator11.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.3/builds/75)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1476)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1502)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1482)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1496)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1481)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1491)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1531)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1471)

**Pull request merged**